ArduPlane/Attitude.cpp

```diff
@@ -377,7 +377,10 @@ void Plane::stabilize()
     }
     float speed_scaler = get_speed_scaler();
 
-    if (quadplane.in_tailsitter_vtol_transition()) {
+    uint32_t now = AP_HAL::millis();
+    // time check deals with a race condition where this function is called before Quadplane has a chance to update the transission state upon entering a VTOL mode
+    const bool tailsitter_vtol_transition = quadplane.in_vtol_mode() && quadplane.is_tailsitter() && ((quadplane.transition_state == QuadPlane::TRANSITION_ANGLE_WAIT_VTOL) || ((now - quadplane.last_vtol_mode_ms) > 1000));
```

Comment:
What's the reason for the hardwired 1000 msec timeout?

Reply:
No reason it has to be 1 second, that is just the same number as the original check here https://github.com/ArduPilot/ardupilot/blob/379506b718bc4719504478d9da979becba666620/ArduPlane/quadplane.cpp#L1839

Reply:
so if you are just entering a VTOL mode after having been non-VTOL for at least 1 second...
I guess the reason is that you want to prevent the transition logic from firing if there's a rapid switch from VTOL to FW and back. 

Seems like this would be easier to understand/maintain if there were some additional states defined. Without drawing a state transition diagram, I can't be sure what's going on.

Reply:
> I guess the reason is that you want to prevent the transition logic from firing if there's a rapid switch from VTOL to FW and back.

That check is what puts you in TRANSITION_ANGLE_WAIT_VTOL in the first place. Its a special case for tailsitters because although the mode instantly changes we really want to stay in forward flight until we get to the transition angle.

The state is already there, just we check it before it gets set correctly